components/eamxx/src/dynamics/homme/physics_dynamics_remapper.cpp

```diff
@@ -205,10 +205,12 @@ update_subfields_views (const std::map<int,SubviewInfo>& subfield_info,
     }
   };
 
-  for (const auto& it : subfield_info) {
-    const auto& f = fields[it.first];
-    if ( not(it.second==f.get_header().get_alloc_properties().get_subview_info()) ){
-      get_view(it.first,fields[it.first]);
+  for (auto [fname, svi]& : subfield_info) {
+    const auto& f = fields[fname];
+    const auto& new_svi = f.get_header().get_alloc_properties().get_subview_info();
+    if ( not(svi==new_svi) ) {
+      get_view(fname,f);
+      svi = new_svi;
```

Comment:
Not exactly sure where this subfield info is used, but seems to make sense to me that this could be a bug.

Reply:
The sv info is used right above, in the if statement, to decide whether we should update the view or not. Honestly, I think we could just _always_ update the views, but we can keep the logic as is. In the P->D remap, otoh, we do NOT allow changing the subview info, so storing it is also needed there.